lib/cylc/remote.py

```diff
@@ -283,3 +298,586 @@ def execute(self, dry_run=False, forward_x11=False, abort_if=None):
             sys.exit(r'ERROR: remote command failed %d' % res)
         else:
             return True
+
+
+class EmptyHostList(Exception):
+    """Exception to be raised if there are no hosts considered valid for
+       appointment given the 'run hosts' and 'run host select' global
+       configuration specifications."""
+
+    def __str__(self):
+        return "No hosts currently compatible with the global configuration."
```

Comment:
I wonder if it is worth printing out what's configured.

Reply:
It would be certainly useful & easy enough to implement, so yes.